JitsiMeetJS.js

```diff
@@ -336,7 +345,25 @@ export default {
                                 reason: 'unsupported resolution'
                             }));
 
-                        return this.createLocalTracks(options);
+                        return this.createLocalTracks(
+                            options,
+                            undefined,
+                            originalOptions || Object.assign({}, options));
+                    }
+
+                    // we tried everything, if there is a mandatory
+                    // device id, remove it and let gum find a device to
+                    // use
+                    if (originalOptions
+                        && error.gum.constraints
+                        && error.gum.constraints
+                        && error.gum.constraints.video
+                        && error.gum.constraints.video.mandatory
+                        && error.gum.constraints.video.mandatory.sourceId
+                            === options.cameraDeviceId) {
```

Comment:
do we actually need to check that it is equal to the one in options?  is there a scenario where it wouldn't be?